sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -131,6 +144,49 @@ protected void beforeTest() {
     protected void afterTest() {
     }
 
+    /**
+     * Get offset which determines the rolling strategy.
+     *
+     * @return The offset according to day of the week and platform information
+     */
+    public static int getOffset() {
+        LocalDate today = LocalDate.now();
+        buildCalendarMap();
+        buildPlatformList();
+        return calendarMap.get(today.getDayOfWeek()) + getPlatFormOffset();
+    }
+
+    private static Integer getPlatFormOffset() {
+        String currentOs = Configuration.getGlobalConfiguration().get(AZURE_TEST_OS_NAME);
+        String currentJdk = Configuration.getGlobalConfiguration().get(AZURE_TEST_JDK_VERSION);
+        return platformList.stream().filter(platform ->
+            currentOs.trim().toLowerCase(Locale.ROOT).contains(platform.split(",")[0].toLowerCase(Locale.ROOT))
+            && currentJdk.trim().toLowerCase(Locale.ROOT).contains(platform.split(",")[1]
+                .toLowerCase(Locale.ROOT))
+        ).map(platformList::indexOf).findFirst().orElse(null);
+    }
+
+    private static void buildPlatformList() {
+        platformList = new ArrayList<>();
```

Comment:
Once this map is not null, you don't need to recreate it on every call. Same with the calendarMap below.

Reply:
Will check null here